File: 3254-find-the-power-of-k-size-subarrays-i/3254-find-the-power-of-k-size-subarrays-i.py
class Solution:
    def resultsArray(self, nums: List[int], k: int) -> List[int]:
        if k>len(nums):
            return [-1]
        if k==len(nums):
            return [nums[-1]]
        if k==1:
            return nums
        res=[]
        for i in range(len(nums)-k+1):
            j=i+1
            while(j<i+k):
                if nums[j-1]==nums[j]-1:
                    if j==i+k-1:
                        res.append(nums[j])
                else:
                    res.append(-1)
                    break
                j+=1
        return res

File: 3254-find-the-power-of-k-size-subarrays-i/3254-find-the-power-of-k-size-subarrays-i.cpp
class Solution {
public:
    vector<int> resultsArray(vector<int>& nums, int k) {
        if (k==1) return nums;
        vector<int> res;
        int n=nums.size();
        int l=0,r=0,score=1;
        while(r<n){
            if(r>0 && nums[r-1]+1==nums[r]) {
                score++;
            }
            if(r-l+1>k){
                if(nums[l]+1==nums[l+1]){
                    score--;
                }
                l++;
            }
            if(r-l+1==k){
                if(score==k) res.push_back(nums[r]);
                else res.push_back(-1);
            }
            r++;
        }
        return res;
    }
};